packages/yew/src/functional/hooks/use_prepared_state/mod.rs

```diff
@@ -0,0 +1,144 @@
+#[cfg(feature = "hydration")]
+pub(super) mod feat_hydration;
+#[cfg(all(feature = "hydration", feature = "ssr"))]
+mod feat_hydration_ssr;
+#[cfg(not(any(feature = "hydration", feature = "ssr")))]
+pub(super) mod feat_none;
+#[cfg(feature = "ssr")]
+mod feat_ssr;
+
+#[cfg(all(feature = "hydration", not(feature = "ssr")))]
+pub use feat_hydration::*;
+#[cfg(all(feature = "ssr", feature = "hydration"))]
+pub use feat_hydration_ssr::*;
+#[cfg(not(any(feature = "hydration", feature = "ssr")))]
+pub use feat_none::*;
+#[cfg(all(feature = "ssr", not(feature = "hydration")))]
+pub use feat_ssr::*;
+/// Use a state prepared on the server side and its value is sent to the client side during
+/// hydration.
+///
+/// The component sees the same value on the server side and client side if the component is
+/// hydrated.
+///
+/// It accepts a closure as the first argument and a dependency type as the second argument.
+/// It returns `SuspensionResult<Option<Rc<T>>>`.
+///
+/// During hydration, it will only return `Ok(Some(Rc<T>))` if the component is hydrated from a
+/// server-side rendering artifact and its dependency value matches.
+///
+/// `let state = use_prepared_state!(|deps| -> ReturnType { ... }, deps)?;`
+///
+/// It has the following signature:
+///
+/// ```
+/// # use yew::prelude::*;
+/// # use serde::de::DeserializeOwned;
+/// # use serde::Serialize;
+/// # use std::rc::Rc;
+/// # use yew::suspense::SuspensionResult;
```

Comment:
These are not a documentation but a function signature:

<img width="967" alt="image" src="https://user-images.githubusercontent.com/11693215/169797486-4c5cec32-b3a5-478e-b68c-34cea4286632.png">

If the imports are shown, it would look like:

<img width="966" alt="image" src="https://user-images.githubusercontent.com/11693215/169797788-d5ae16b8-31f9-4e8e-9a28-d0c8c3fde3db.png">

Which way looks better?

Reply:
In the function signature, rustdoc hyperlinks the types so it's clear what type it exactly is. We don't have that ability here so we should show the imports. If you want to hide the `std` and `serde` imports, that's fine but I would like to at least show the `yew` imports